### DISCUSSION TITLE
Module Federation within Turbopack

### DISCUSSION THREAD

**Discussion by slinkardbrandon** [vercel/next.js#69836] [Ideas]
Hello! I watched part of the turbopack announcement today at Next conf and was immediately very excited to hear about it (thanks!).

I have a few projects that I maintain today that are built leveraging [Webpack module federation](https://webpack.js.org/concepts/module-federation/). Is this something that may be supported by turbopack now or in the future?

I think it could be very interesting to be able to achieve federated components for use within microfrontends with NextJS and turbopack so I figured I'd ask / contribute the idea if it hasn't been already. That said, I know the tool is in alpha so I acknowledge this is quite early on to propose / ask about this.

If this is something that's being discussed, I'd love to help 😄 

Thanks for the awesome tools!

**Comment by nate-summercook** [comment]
I second that. Also working on a bigger microfrontend project leveraging webpack module federation. If turbopack would support such a feature, that would be a big reason to consider the switch :)

**Reply by Mboulianne** [reply]
+1

**Comment by adroste** [comment]
If you (as a large and trustworthy company) are brave enough for this kind of marketing:

> We've built Turbopack as the successor of Webpack - the bundler that's at the heart of many beloved frameworks. Any successor to Webpack would need to be faster and just as flexible and extensible.

*src:* [Turbo.build - Turbopack Vs Webpack - Web Archive 25.10.22](https://web.archive.org/web/20221025200228/https://turbo.build/pack/docs/comparisons/turbopack-vs-webpack)

the question should rather be "when" instead of "if" 😉.

So if a lot of people vote for this, maybe we'll have it sooner. 🎉

**Comment by nandeshwarshubh** [comment]
We in our company as well are using webpack Module Federation, the only thing stopping us from switching to turbopack is support for MF.

**Comment by yordan-kanchelov** [comment]
It's kinda surprising to see not much interest around this feature in this thread. 

Any chance the Turbopack team could share if this is being considered for future releases? A little insight would be cool and might spark more discussion here.

**Comment by amir2mi** [comment]
Actually using monorepo that Turborepo offers is a big and even unnecessary move for some projects, most of the time Module Federation would suffice. For example, you need to have a host as a wrapper UI and some apps inside.
Just thinking about having Turbopack and MF together would be a great deal.

**Comment by hansfpc** [comment]
Any plans for MF on Turbopack?

**Comment by raolak** [comment]
+1 for MF on Turbopack.  Any plans to support this? This is one of the key requirement for my project !

**Comment by vikram-shyftlabs** [comment]
Any update on the same

**Comment by lukovskij** [comment]
do you have any updates?

**Comment by lukovskij** [comment]
do you have any updates for MF on Turbopack?

**Comment by maggiben** [comment]
Im leveraging module federation with nextjs, currently Im very happy with the combination of `@module-federation/nextjs-mf` and `next` the results have ben very positive so far, but the rest of the frontend teams in the company are skeptical for the reasons discussed here

**Reply by Netail** [reply]
FYI; https://x.com/ScriptedAlchemy/status/1846324235800641735 (from the creator of moduled federation)

**Comment by akifunal** [comment]
Are there any plans to support MF in Turbopack with the App Router?

**Reply by zackarychapple** [reply]
Starting to see more discussions on this lately. Stay tuned!

**Comment by mo-hawary** [comment]
Any update on this? and how can we help?

**Reply by zackarychapple** [reply]
There is some underlying work happening right now to support Module Federation with Next (which is a pre-curser to this). Hopefully will have some progress on that publicly in a few weeks. I'd assume that port to Turbopack will be a follow up after that.

**Reply by jubinpyli** [reply]
hey @zackarychapple - any progress on this?

**Comment by rvveber** [comment]
It's been a while, any update on compatibility with Turbopack? And how can we help?

**Reply by Netail** [reply]
In a [recent podcast](https://youtu.be/tgfjhAy85UI) they did mention they are working on a concept called "remote components", which they are testing internally & with partners;

- [NPM package](https://www.npmjs.com/package/remote-components)
- [Demo app 1](https://github.com/vercel-labs/remote-components-nextjs-app)
- [Demo app 2](https://github.com/vercel-labs/remote-components-in-mdx)

I assume that will support Turbopack
